ex01.exs

```diff
@@ -74,14 +78,24 @@ end
 defmodule Ex01 do
 
   def counter(value \\ 0) do
-    # ...your code
+    receive do
+      {:next, from}  ->
+        send from, {:next_is, value}
+        counter(value + 1)
+      {:next_is, value} ->
+        counter(value)
+    end
   end
 
   def new_counter(initial_value \\ 0) do
-    # ... your code
+    apiCount = spawn Ex01, :counter, [initial_value]
```

Comment:
This line generates a warning, because `apiCount` is never used.

Treat warnings like errors, and never be happy with code that generates them. In my experience, debugging sessions often end at lines that the compiler flagged with a warning.

# -1